app/validators/schema_validator.rb

```diff
@@ -0,0 +1,52 @@
+require "json-schema"
+
+class SchemaValidator
+  def initialize(rendered_content_item)
+    @rendered_content_item = rendered_content_item.deep_symbolize_keys
+  end
+
+  def validate
+    return true if schema_name_exception?
+    validate_schema
+  end
+
+private
+
+  attr_reader :rendered_content_item
+
+  def validate_schema
+    JSON::Validator.validate!(schema, rendered_content_item)
+  rescue JSON::Schema::ValidationError => error
+    Airbrake.notify_or_ignore(error, parameters: {
```

Comment:
I've had a chat with @boffbowsh.  Airbrake's async stuff is OK-ish but we're going to look into using github scientist for this (and another thing) since we've been wanting to try it out.